Source/WTF/wtf/FileSystem.cpp

```diff
@@ -453,15 +453,17 @@ MappedFileData mapToFile(const String& path, size_t bytesSize, Function<void(con
         return { };
     }
 
-    void* map = const_cast<void*>(mappedFile.data());
-    uint8_t* mapData = static_cast<uint8_t*>(map);
+    if (outputHandle)
+        *outputHandle = handle;
```

Comment:
I wish we had an RAII class that represented a single owner of PlatformFileHandle, and closed the file in its destructor. Then we wouldn't need all the closeFile() calls above, and we could do a WTFMove(handle) here.

Reply:
That would indeed be a nice addition.